src/Peachpie.Runtime/Attributes.cs

```diff
@@ -287,6 +287,26 @@ public sealed class NotNullAttribute : Attribute
 
     }
 
+    /// <summary>
+    /// Compiler generated attribute denoting that the method returns a specific type
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class ReturnTypeAttribute : Attribute
```

Comment:
I think this is redundant - we could rely on .NET type. Annotating every method with an attribute and using too much strings may not worth it.